source/DistanceMap.cpp

```diff
@@ -309,7 +310,8 @@ bool DistanceMap::CheckLink(const System *from, const System *to, bool useJump)
 	// the two systems that you can jump between them, you can plot a course
 	// between them even if neither system is explored. Otherwise, you need to
 	// know if a link exists, so you must have explored at least one of them.
-	if(useJump && from->Position().Distance(to->Position()) <= System::NEIGHBOR_DISTANCE)
+	double distance = from->JumpRange() ? from->JumpRange() : jumpRange;
```

Comment:
The code here seems fine, but add some comment here to explain that if the from system has a JumpRange, then we use the source-systems jump-range, otherwise we use the ships jump-range.